### ISSUE TITLE
Allow users to select characteristics when creating new host

### ISSUE THREAD

**Issue by ageorgou** [UCL-CloudLabs/user-portal#48]
We should give users some flexibility in creating a VM. This could include presenting them with options concerning operating system, size of the machine etc.  in the form for adding a new host, with the consequent changes in the model and the terraform script.

**Comment by raquelalegre** [comment]
Some progress so far on the deployment side as part of #72. Model updated as well as TF template. Tests updated accordingly as well and they pass for Ubuntu machines no problem.

I'm now working on parametrizing the tests to try the different combinations of vm type and os that we'll give to users in the front end. 

Seems like Windows machines are hard to deploy and we'll need to make the terraform template more modular, or have one for Windows VMs and one for Linux VMs. This constraints is due to the submodule called `os_profile_linux_config` in the TF template that we have been using so far. The Windows equivalent `os_profile_windows_config` is completely different.

https://www.terraform.io/docs/providers/azurerm/r/virtual_machine.html#os_profile_windows_config

Once I sort this out, I'll update the form in the front end as well.

**Comment by raquelalegre** [comment]
Also, the tests are currently a bit overcomplicated for what we need, so I'll try and refactor a little bit.

I also would like to handle the OS separately as a class instead of as 4 new properties of the host (publisher, offer, sky and version). If I have time...

**Comment by raquelalegre** [comment]
Plus all the bit with the script to run the docker-sample repo 😱

**Comment by raquelalegre** [comment]
For now I'm stuck at the ssh configuration for windows machines. I shall read more about that. Here's what I was about to read next: http://blog.superautomation.co.uk/2016/11/configuring-terraform-to-use-winrm-over.html, https://github.com/hashicorp/terraform/issues/14953 and https://www.reddit.com/r/devops/comments/78yfns/how_to_get_terraform_remoteexec_to_successfully/. And maybe https://github.com/dhoer/terraform_examples. 

Also this might help for templating terraform a bit for the ubuntu/windows options https://www.terraform.io/docs/providers/template/index.html.

**Comment by raquelalegre** [comment]
I managed to successfully launch a Windows VM with Terraform and but configuring WinRM to connect to it is very cumbersome and would require other changes to the portal (e.g. need to configure a certificate to establish secure connections to the machine in a manner similar to the ssh keys, so this implies changes in the profile configuration and the new host form, as well as the corresponding validation - can't choose a windows machine and authentication with ssh keys).

Other cons:
* Launching the Windows VM in a comparable machine takes between 4x and 6x longer than the Ubuntu equivalent.
* Windows VMs are more expensive, which leaves users with less allowance within the free tier.
* We have no users that have expressed they require Windows VMs - they have shown interest, but they can all work with (and prefer) Unix machines.
* I haven't gotten to the point of running the commands to install everything to be able to run docker. This might also be very cumbersome.

We can come back to configure Windows machines when we have users asking for this feature in the future, but at the moment we are short on development time and I've wasted too long trying to configure it already. 

OS options will be different Ubuntu versions and maybe other Unix-like OS, depending on what time allows.
